### ISSUE TITLE
Sanitize character names

### ISSUE THREAD

**Issue by GrimKriegor** [TES3MP/TES3MP#17]
On yesterday and today's test sessions it became apparent character names with leading or trailing white-spaces as well as non-alphanumeric characters would cause unexpected behavior.

I have tried patching that in the server scripts, example:

```
local pname = tes3mp.GetName(pid)
pname = pname:gsub('%W','') -- Remove all non-alphanumeric characters
pname = pname:gsub("^%s*(.-)%s*$", "%1") -- Remove leading and trailing white-spaces
Players[pid].name = pname
```

But it would be a much more elegant and reliable solution to take care of this problem at the source so we get a clean and safe string whenever the user logs in or we call tes3mp.GetName(). What do you gentlemen think?

**Comment by Koncord** [comment]
IMO this should be resolved by scripts

**Comment by davidcernat** [comment]
A simple Regex check should probably be enough on the C++ side though.

**Comment by Koncord** [comment]
IMO in the server side we should accept only alphanumeric & spacers. But i don't know how it should works with i18n.

**Comment by davidcernat** [comment]
Maybe allow the same characters as Windows does for filenames?

**Comment by Koncord** [comment]
Do you mean just ignore [not allowed](https://msdn.microsoft.com/en-gb/library/windows/desktop/aa365247%28v=vs.85%29.aspx#naming_conventions) characters?

**Comment by davidcernat** [comment]
Yep.

**Comment by davidcernat** [comment]
Indecision about this matter was leading to server crashes, so I've provided a fix with TES3MP/PluginExamples@398ff45fb5c134bc40e17b3211f600b595c6c345.
